crates/rolldown/src/utils/pre_process_ecma_ast.rs

```diff
@@ -74,26 +73,9 @@ impl PreProcessEcmaAst {
       || !matches!(bundle_options.target, ESTarget::EsNext)
     {
       let ret = ast.program.with_mut(|fields| {
-        let transform_options = if matches!(parsed_type, OxcParseType::Js) {
-          Cow::Borrowed(&bundle_options.transform_options)
-        } else {
-          let mut transformer_options = bundle_options.transform_options.clone();
-          let jsx_plugin = match &bundle_options.jsx {
-            Jsx::Disable => unreachable!("Jsx::Disable should be failed at parser."),
-            // The oxc jsx_plugin is enabled by default, we need to disable it.
-            Jsx::Preserve => false,
-            Jsx::Enable(jsx) => {
-              transformer_options.jsx =
-                NormalizedBundlerOptions::merge_jsx_options(jsx.clone(), transformer_options.jsx);
-              matches!(parsed_type, OxcParseType::Tsx | OxcParseType::Jsx)
-            }
-          };
-
-          transformer_options.jsx.jsx_plugin = jsx_plugin;
-          Cow::Owned(transformer_options)
-        };
-
-        Transformer::new(fields.allocator, Path::new(path), &transform_options)
+        let transform_options = &bundle_options.transform_options;
+
+        Transformer::new(fields.allocator, Path::new(path), transform_options)
```

Comment:
The configuration merging is finished here https://github.com/rolldown/rolldown/pull/3854/files#diff-9c7476e5ae8a04a7e3a91c4f26589038b8e25d9fc08213b21b3683a63580daccR182-R190, so it did not need to merge every thing.